dsl.go

```diff
@@ -343,6 +357,8 @@ func (matchers Matchers) Matches(params []Param) bool {
 }
 
 func (matchers *Matchers) append(matcher Matcher) {
+	matchersMutex.Lock()
+	defer matchersMutex.Unlock()
```

Comment:
Locking not needed, if we require that setting up mocks happens strictly in a non-concurrent fashion.